rosbag2_cpp/test/rosbag2_cpp/test_multifile_reader.cpp

```diff
@@ -114,3 +113,95 @@ TEST_F(MultifileReaderTest, get_all_topics_and_types_throws_if_no_storage)
 {
   EXPECT_ANY_THROW(reader_->get_all_topics_and_types());
 }
+
+class FakeSequentialReader : public rosbag2_cpp::readers::SequentialReader
```

Comment:
Maybe use `MockSequentialReader` instead of `FakeSequentialReader` to keep it consistent with rest of the tests..

Reply:
Since this is implementing a reader and not mocking it (i.e. using Gmock) it is technically a fake.
Check [this SO](https://stackoverflow.com/questions/346372/whats-the-difference-between-faking-mocking-and-stubbing) answer and this [article](https://www.martinfowler.com/articles/mocksArentStubs.html)